### PR TITLE
Merchant stats#94

### LINKED ISSUE #94: User Story 40, Merchant Dashboard Statistics
```
As a merchant
When I visit my dashboard, I see an area with statistics:
- top 5 items I have sold by quantity
- total quantity of items I've sold, and as a percentage against my sold units plus remaining inventory (eg, if I have sold 1,000 things and still have 9,000 things in inventory, the message would say something like "Sold 1,000 items, which is 10% of your total inventory")
- top 3 states where my items were shipped
- top 3 city/state where my items were shipped (Springfield, MI should not be grouped with Springfield, CO)
- name of the user with the most orders from me (pick one if there's a tie)
- name of the user who bought the most total items from me (pick one if there's a tie)
- top 3 users who have spent the most money on my items
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #97

depends on #96

### PR DESCRIPTION
**Pull request by le3ah**
All tests passing; closes #94

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -97,10 +97,32 @@ def pending_orders
...
+    end
+    def top_shipment_city_states
+      User.joins("INNER JOIN orders ON orders.user_id = users.id INNER JOIN order_items ON order_items.order_id = orders.id INNER JOIN items ON order_items.item_id = items.id")
+      .select("users.city, users.state, SUM(order_items.quantity) as order_item_quantity")
+      .where("orders.status = 'complete'")
+      .where("items.user_id=#{self.id}")
+      .group("users.city, users.state")
+      .order("order_item_quantity")
+      .limit(3)
+    end
```
Comment (stoic-plus) [inline app/models/user.rb]: I would put these into a separate method that takes options for what to select and group by (look at my `top_cities_or_states`) but it's not that important. Great job!

File: app/models/user.rb [hunk 2]
```diff
@@ -119,28 +119,31 @@ def top_shipment_city_states
...
     end
     def top_user_spenders
       User.joins("INNER JOIN orders ON orders.user_id = users.id INNER JOIN order_items ON order_items.order_id = orders.id INNER JOIN items ON items.id=order_items.item_id")
-      .select("orders.user_id AS user_id, sum(order_items.quantity * order_items.price) as customer_spend_total")
+      .select("users.*, sum(order_items.quantity * order_items.price) as customer_spend_total")
       .where("items.user_id=#{self.id}")
       .where("order_items.fulfilled=true")
-      .group("orders.user_id")
+      .group("users.id")
       .order("customer_spend_total desc")
```
Comment (stoic-plus) [inline app/models/user.rb]: I was curious if this would work. It's great that you implemented this.
